### PR TITLE
Reduce redundant else-branching (after throws), add missing @throws annotations

### PR DESCRIPTION
**Pull request by kstenschke**

### CODE REVIEW

File: library/Zend/Amf/Server.php [hunk 1]
```diff
@@ -294,10 +297,13 @@ protected function getLoader()
      * Loads a remote class or method and executes the function and returns
      * the result
      *
-     * @param  string $method Is the method to execute
-     * @param  mixed $param values for the method
+     * @param string $method Is the method to execute
+     * @param null $params
```
Comment (Shardj) [inline library/Zend/Amf/Server.php:301]: Params is mixed nullable with a default of null, same for source. I would put it as mixed or mixed|null

Comment (kstenschke) [reply library/Zend/Amf/Server.php:301]: Though i agree that declaring nullability explicitly might be helpful, to my understanding, null is one of the types included in "mixed". 
See: https://wiki.php.net/rfc/mixed-typehint

Comment (Shardj) [reply library/Zend/Amf/Server.php:301]: Fair enough, feel free to change it to just mixed then

**Comment by kstenschke** [comment]
was that finding accidental or after running some more thorough inspection?
(if accidental, i will re-check the PR, to ensure avoiding other flaws)

**Comment by Shardj** [comment]
I gave the change a quick skim through, so accidental. However once I found that one I did look for others and couldn't see any further problems

I'm not planning on merging with #96 at all

**Comment by kstenschke** [comment]
Thanks for your feedback. 
Wiith a less redundant code base, further enhancements should go easier, therefor i'd ask you to merge.
I also had a look at #96 / zfs, and think if they should decide on merging from zf1-future, also that shouldn't pose a problem.

**Comment by kstenschke** [comment]
Thank you for merging 👍

File: library/Zend/Validate/File/MimeType.php [hunk 2]
```diff
@@ -161,6 +161,7 @@ public function __construct($mimetype)
      * magic file.
      *
      * @return string
+     * @throws Zend_Validate_Exception
```
Comment (glensc) [inline library/Zend/Validate/File/MimeType.php:164]: this throws is bogus, it's never thrown (but caught and ignored silently)

will be removed when porting https://github.com/zf1s/zf1/pull/54

Comment (glensc) [reply library/Zend/Validate/File/MimeType.php:164]: https://github.com/Shardj/zf1-future/pull/265

File: library/Zend/Cloud/QueueService/Adapter/ZendQueue.php [hunk 3]
```diff
@@ -147,9 +152,10 @@ public function listQueues($options = null)
...
      * Get a key/value array of metadata for the given queue.
      *
-     * @param  string $queueId
-     * @param  array  $options
+     * @param string $queueId
+     * @param array $options
      * @return array
+     * @throws Zend_Cloud_QueueService_Exception
      */
     public function fetchQueueMetadata($queueId, $options = null)
```
Comment (glensc) [inline library/Zend/Cloud/QueueService/Adapter/ZendQueue.php:160]: this method also returns `false`.

File: library/Zend/Amf/Server.php [hunk 4]
```diff
@@ -709,8 +722,9 @@ public function getRequest()
     /**
      * Public access method to private Zend_Amf_Server_Response reference
      *
-     * @param  string|Zend_Amf_Server_Response $response
+     * @param string|Zend_Amf_Server_Response $response
```
Comment (glensc) [inline library/Zend/Amf/Server.php:725]: Zend_Amf_Server_Response class does not exist:
- https://github.com/zf1s/zf1/pull/137/files

**Pull request merged**